### PR TITLE
Fix stepper: no precheck for undefined variables

### PR DESCRIPTION
**Pull request by Ziwen510**
Add a checker for undefined variables in the stepper before the interpreter is run.

### CODE REVIEW

File: src/stepper/converter.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ function evaluateFunctionObject(node: substituterNodes, context: Context) {
 }
 
 export function objectToString(value: any): string {
-  if ('toReplString' in value) {
+  if (!(value === null) && 'toReplString' in value) {
```
Comment (RichDom2185) [inline src/stepper/converter.ts:94]: Can we use `value !== null` instead for simplicity?

File: src/stepper/stepper.ts [hunk 2]
```diff
@@ -2501,7 +2508,7 @@ function jsTreeifyMain(
     },
 
     Literal: (target: es.Literal): es.Literal => {
-      if (typeof target.value === 'object') {
+      if (typeof target.value === 'object' && !(target.value === null)) {
```
Comment (RichDom2185) [inline src/stepper/stepper.ts:2511]: Likewise, can we use `target.value !== null` here?

**Review by RichDom2185 — COMMENTED** [review]
Thanks for working on this so quickly! Is it possible to add more test cases in order to keep the test coverage up? What was the reason that some of the original test cases were deleted?

**Review by martin-henz — APPROVED** [review]
Looks good!

**Comment by Ziwen510** [comment]
The three deleted test cases will actually cause undefined variable error in our normal transpiler run, this pr makes them raise errors in stepper (which is originaly runnable). I have added one new test case.

File: src/stepper/converter.ts (hunk 1 above)
Comment (Ziwen510) [reply src/stepper/converter.ts]: Done

File: src/stepper/stepper.ts (hunk 2 above)
Comment (Ziwen510) [reply src/stepper/stepper.ts]: Done

**Comment by RichDom2185** [comment]
> The three deleted test cases will actually cause undefined variable error in our normal transpiler run, this pr makes them raise errors in stepper (which is originaly runnable). I have added one new test case.

Hmm, perhaps we can leave this as an issue then, cause it makes sense to test each individual behaviour separately.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Comment by RichDom2185** [comment]
> > The three deleted test cases will actually cause undefined variable error in our normal transpiler run, this pr makes them raise errors in stepper (which is originaly runnable). I have added one new test case.
> 
> Hmm, perhaps we can leave this as an issue then, cause it makes sense to test each individual behaviour separately.

Referenced in #1475.

**Pull request merged**